lib/manager/components/transform/FeedTransformation.js

```diff
@@ -0,0 +1,214 @@
+      default:
+        break
+    }
+    return fields
+  }
+
+  _getValidationIssues = () => {
+    const {feedSource, transformation} = this.props
+    // Get selected version (if applicable).
+    const version = feedSource.feedVersions &&
+      feedSource.feedVersions.find(v => v.id === transformation.sourceVersionId)
+    const issues = []
+    if (!transformation.table) {
+      issues.push('Table must be defined')
+    }
+    switch (transformation['@type']) {
+      case 'ReplaceFileFromVersionTransformation':
+        // Only trigger validation issue if versions have been loaded.
+        if (!version && feedSource.feedVersions) {
+          issues.push('Version must be defined')
+        }
+        break
+      case 'ReplaceFileFromStringTransformation':
+        if (!transformation.csvData) {
+          issues.push('CSV Data must be defined')
+        }
+        // TODO: Further validate CSV data based on parsing fields, checking for
+        //   missing columns, etc.
+        break
+      default:
+        break
+    }
+    return issues
+  }
+
+  _onChangeCsvData = (evt: SyntheticInputEvent<HTMLInputElement>) => {
+    this.setState({csvData: evt.target.value})
+  }
+
+  _onSaveCsvData = () => {
+    const csvData = this.state.csvData || null
+    this.setState({csvData: null})
+    this.props.onChange({csvData}, this.props.index)
+  }
+
+  _onRemoveTransformation = () => {
+    this.props.onRemove(this.props.index)
+  }
+
+  _onSelectTable = (tableOption: ReactSelectOption) => {
+    this.props.onChange({table: tableOption.value}, this.props.index)
+  }
+
+  _onSelectVersion = (versionIndex: number) => {
+    const {feedSource, onChange} = this.props
+    if (feedSource.feedVersions) {
+      const version = feedSource.feedVersions[versionIndex - 1]
+      onChange({sourceVersionId: version.id}, this.props.index)
+    } else {
+      console.warn('Feed source does not have list of feed versions')
+    }
+  }
+
+  render () {
+    const {feedSource, index, transformation} = this.props
+    const tables = [...getGtfsSpec()].filter(t => !t.datatools)
+    if (isModuleEnabled('gtfsplus')) tables.push(...getGtfsPlusSpec())
+    const transformationType = transformation['@type']
+    const validationIssues = this._getValidationIssues()
+    const backgroundColor = validationIssues.length === 0
+      ? '#f7f7f7'
+      : '#ffdf68'
+    return (
+      <div style={{
+        backgroundColor,
+        padding: '2px 10px 10px 10px',
+        marginBottom: '10px',
+        border: '#ccc 1px solid',
+        borderRadius: '10px'
+      }}>
+        <h5>
+          <Button
+            bsSize='xsmall'
+            className='pull-right'
+            onClick={this._onRemoveTransformation}>
+            Delete step
+          </Button>
+          Step {index + 1} - {getTransformationName(transformationType, transformation, feedSource.feedVersions)}
+        </h5>
+        <div className='feed-transformation-body'>
+          <Select
```

Comment:
Why does this drop down appear grayed out and the version selector one appear normal?

Reply:
It's just a different type of selector with a placeholder. The version one is a button selector, this is a select component that allows for text entry.